### PR TITLE
sqlbase: add randomized tests for key and value encoding

### PR DESCRIPTION
**Pull request by jordanlewis**
This uses `gopter`, a property-based testing framework. More of a POC of how that framework works than anything, though I don't think we have tests that do this work yet so it'll be good to add in any event.

Release note: None

### CODE REVIEW

**Comment by jordanlewis** [comment]
Ping. @Mji

**Comment by jordanlewis** [comment]
Last comment was sent hastily - ping @mjibson, could you take a look please?

File: pkg/sql/sqlbase/column_type_encoding_test.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			false), gopter.NoShrinker)
+	}
+}
+
+func genDatumWithType(columnType interface{}) gopter.Gen {
+	return func(genParams *gopter.GenParameters) *gopter.GenResult {
+		datum := RandDatum(genParams.Rng, columnType.(ColumnType),
+			false)
+		if datum == nil {
+			fmt.Println("WOT M8", columnType)
```
Comment (mjibson) [inline pkg/sql/sqlbase/column_type_encoding_test.go]: should this be here?

**Review by madelynnblue — APPROVED** [review]

**Pull request merged**